test/test_datapipe.py

```diff
@@ -829,30 +831,30 @@ def test_fork_iterdatapipe(self):
         self.assertEqual(list(range(5)), output2)
         self.assertEqual(list(range(10)), output3)
 
-        # Reset Test: DataPipe doesn't reset if this pipe hasn't been read
+        # Reset Test: DataPipe resets when a new iterator is created, even if this datapipe hasn't been read
         dp1, dp2 = input_dp.fork(num_instances=2)
-        i1, i2 = iter(dp1), iter(dp2)
+        _ = iter(dp1)
         output2 = []
-        for i, n2 in enumerate(i2):
-            output2.append(n2)
-            if i == 4:
-                i1 = iter(dp1)  # Doesn't reset because i1 hasn't been read
-        self.assertEqual(list(range(10)), output2)
+        with self.assertRaisesRegex(RuntimeError, r"iterator has been invalidated"):
+            for i, n2 in enumerate(dp2):
+                output2.append(n2)
```

Comment:
snit: replace next 3 lines with `pass`

Reply:
The invalidation happens in the middle of the loop during the `if` condition. Therefore, it cannot be changed to `pass` and therefore will still be elements in `output2` that can be checked against.